Document Cache-Control (#958)
cap-js/docs: advanced/fiori.md

:::

## Cache Control

> To indicate that caches can store the response and reuse it for subsequent requests, CAP provides the option to set a `Cache-Control: max-age=<seconds>` HTTP header. This can be done via `@http.CacheControl: {maxAge: <seconds>}` CDS annotation on stream properties. The header allows to control the behavior of caches, the `max-age` (in seconds) specifies the maximum age of the content before it becomes stale.

Reviewer: Maybe, a bit close to the Mozilla docu:

```suggestion
CAP provides the option to set a [Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) header with a [max-age](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#max-age) directive to indicate that a response remains fresh until _n_ seconds after it was generated .
In the CDS model, this can be done via a `@http.CacheControl: {maxAge: <seconds>}` annotation on stream properties. The header indicates that caches can store the response and reuse it for subsequent requests while it's fresh.
The `max-age` (in seconds) specifies the maximum age of the content before it becomes stale.
```

The max-age=N response directive indicates that the response remains [fresh](https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching#fresh_and_stale_based_on_age) until N seconds after the response is generated.

:::

## Cache Control

CAP provides the option to set a [Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) header with a [max-age](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#max-age) directive to indicate that a response remains fresh until _n_ seconds after it was generated .
In the CDS model, this can be done via a `@http.CacheControl: {maxAge: <seconds>}` annotation on stream properties. The header indicates that caches can store the response and reuse it for subsequent requests while it's fresh.
The `max-age` (in seconds) specifies the maximum age of the content before it becomes stale.

:::info Note
The `max-age` is the elapsed time since the response was generated on the origin server (not the elapsed time since the response was received).
:::

> 

Reviewer: I think we need to mention that this feature is available in CAP Java only.

Author: Currently yes, but the Node Stack also plans to support it. So far there was ONLY one team, who requested this feature.

:::

## Cache Control

CAP provides the option to set a [Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) header with a [max-age](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#max-age) directive to indicate that a response remains fresh until _n_ seconds after it was generated .

> In the CDS model, this can be done via a `@http.CacheControl: {maxAge: <seconds>}` annotation on stream properties. The header indicates that caches can store the response and reuse it for subsequent requests while it's fresh.

Reviewer 2: ```suggestion
In the CDS model, this can be done using the `@http.CacheControl: {maxAge: <seconds>}` annotation on stream properties. The header indicates that caches can store the response and reuse it for subsequent requests while it's fresh.
```

The `max-age` (in seconds) specifies the maximum age of the content before it becomes stale.

> :::info Note

Reviewer 2: ```suggestion
:::info Elapsed time since the response was generated
```

> The `max-age` is the elapsed time since the response was generated on the origin server (not the elapsed time since the response was received).

Reviewer 2: ```suggestion
The `max-age` is the elapsed time since the response was generated on the origin server. It's not related to when the response was received.
```
